unit_test/solver/matrix/test_equation_object.py

```diff
@@ -0,0 +1,64 @@
+#  Copyright (c) 2023. Deltares & TNO
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Test equation object."""
+import unittest
+
+import numpy as np
+
+from simulator_core.solver.matrix.equation_object import EquationObject
+
+
+class EquationObjectTest(unittest.TestCase):
+    """Test the equation object."""
+
+    def test_init(self):
+        """Test the init of the equation object."""
+        # arrange
+
+        # act
+        equation_object = EquationObject()  # act
+
+        # assert
+        self.assertEqual(equation_object.coefficients.size, 0)
+        self.assertEqual(equation_object.indices.size, 0)
+        self.assertEqual(equation_object.rhs, 0.0)
+
+    def test_set_coefficient(self):
+        """Test the set coefficient of the equation object."""
+        # arrange
+        equation_object = EquationObject()
+
+        # act
+        #  to discuss is this the preferred way of setting the coefs, or should
+        #  we add it to the constructor?
+        equation_object.coefficients = np.array([1.0])  # act
+
+        # assert
+        self.assertEqual(equation_object.coefficients.size, 1)
+        self.assertEqual(equation_object.coefficients[0], 1.0)
+
+    def test_to_list(self):
+        """Test the to list of the equation object."""
+        # arrange
+        equation_object = EquationObject()
+        equation_object.coefficients = np.array([1.0])
+        equation_object.indices = np.array([1])
+
+        # act
+        result = equation_object.to_list(2)
```

Comment:
Wat betekend de length (d.w.z. length=2) in deze methode? Hoe wordt dit gebruikt?

Als de "length" kleiner is dan het aantal coefficienten maar je loopt er wel over gaat dit dan goed?

Reply:
Dit werkt niet, je zal een error handling toe moeten voegen die checkt op het aantal/lengte van de opgegeven vergelijkingen. 

```suggestion
    def to_list(self, length: int) -> list[float]:
        """Method to change the equation object to list which can be stored in the matrix.

        The method creates a list with zero of the length given in the input.
        At the indices of the class the coefficients of the class are stored.
        :param int length: Length of the list to be returned
        :return: a list of the coefficient which can be used in the matrix.
        """
        if length >= len(self.coefficients):
            coefficient_list: list[float] = [0.0] * length
            for col, coefficient in zip(self.indices, self.coefficients):
                coefficient_list[col] = coefficient
            return coefficient_list
        else:
            raise IndexError("Length of {length} smaller than the available number of coefficients {len(self.coefficients)}!)

```

Reply:
Ik heb het aangepast